### PR TITLE
workaround for assertion failure

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/14711

In contrast to what the branch name indicates, this is actually not (yet) a bugfix. The PR adds more assertions to the WAL tailing and a helper function to print write batch sequence numbers in maintainer mode. This can be used to debug the assertion failure reported in BTS-564

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

#### Related Information

- [x] GitHub issue / Jira ticket number: https://arangodb.atlassian.net/browse/BTS-564

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*

### CODE REVIEW

File: tests/js/client/server_parameters/test-statistics-enabled-false.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+////////////////////////////////////////////////////////////////////////////////
+
+if (getOptions === true) {
+  return {
+    'server.statistics': "false",
+    'server.export-metrics-api': "true",
+  };
+}
+
+const jsunity = require('jsunity');
```
Comment (mpoeter) [inline tests/js/client/server_parameters/test-statistics-enabled-false.js:38]: How is this file related to the changes?

File: arangod/RocksDBEngine/RocksDBWalAccess.cpp [hunk 2]
```diff
@@ -777,6 +777,95 @@ class MyWALDumper final : public rocksdb::WriteBatch::Handler, public WalAccessC
...
+
+    dumper.startNewBatch(batch.sequence);
+    s = batch.writeBatchPtr->Iterate(&dumper);
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+    if (batch.writeBatchPtr->Count() == 0) {
+      // there can be completely empty write batches. in case we encounter
+      // some, we cannot assume the tick gets increased next time
+      dumper.disableTickCheck();
+    }
+#endif
```
[the remark this answers is not in the document]
Comment (jsteemann) [reply arangod/RocksDBEngine/RocksDBWalAccess.cpp]: Fixed.

File: tests/js/client/server_parameters/test-statistics-enabled-false.js (hunk 1 above)
Comment (jsteemann) [reply tests/js/client/server_parameters/test-statistics-enabled-false.js]: It wasn't. But somehow the PR was branched off a wrong version of 3.8, with some changes on top. These changes have now been merged into 3.8, so the PR is now as small as originally intended.

**Review by mpoeter — APPROVED** [review]
